tests/system/small/test_session.py

```diff
@@ -309,6 +309,64 @@ def test_read_gbq_w_script_no_select(session, dataset_id: str):
     assert df["statement_type"][0] == "SCRIPT"
 
 
+@pytest.mark.parametrize(
+    ("query_or_table", "filters", "validator"),
+    [
+        pytest.param(
+            """SELECT
+                rowindex,
+                string_col,
+            FROM `{scalars_table_id}` AS t
+            """,
+            [("rowindex", "<", 4), ("string_col", "==", "Hello, World!")],
+            lambda row: row["rowindex"] < 4 and row["string_col"] == "Hello, World!",
+            id="query_input",
+        ),
+        pytest.param(
+            "{scalars_table_id}",
+            [("date_col", ">", "2022-10-20")],
+            lambda row: pd.to_datetime(row["date_col"]) > pd.to_datetime("2022-10-20"),
+            id="table_input",
+        ),
+        pytest.param(
+            "{scalars_table_id}",
+            [
+                (("rowindex", "not in", [0, 6])),
```

Comment:
double parentheses here is effectively same as single parentheses, did you mean it to represent a test case with `Iterable[Iterable[Tuple]]` filter type, then need to add a comma - `(("rowindex", "not in", [0, 6]),)`